test/array_collapsing_test.dart

```diff
@@ -87,8 +86,6 @@ void main() {
       Vector({'a': 123}, {'b': 123}),
     ];
     await SimCompare.checkFunctionalVector(mod, vectors);
-    SimCompare.checkIverilogVector(mod, vectors);
```

Comment:
this unpacked fails with iverilog? just checking

Reply:
actually, does it fail in iverilog due to tool support or is it illegal in SystemVerilog?  If it's illegal, then we should be doing a check to *not* merge assignments for unpacked arrays instead of just removing this test.

Reply:
yes it fails

Reply:
Hmm.. I haven't checked if if it's illegal. I was just under the impression that its not supported in iverilog. Will not merge until I make sure.

Reply:
looks like it IS legal from LRM 7.4.3 to do unpacked to unpacked assignments